src/session/index.js

```diff
@@ -17,8 +17,9 @@ const redisOrInMemory = (options = {}) => {
 const sessionOptions = (userOpts, store, req) => {
   const userCookie = userOpts.cookie || {};
   const cookie = Object.assign({}, {
-    secure: defaultIfUndefined(userCookie.secure, !isTest),
-    expires: defaultIfUndefined(userCookie.expires, false)
+    secure: req.secure,
```

Comment:
Was the change to the `secure` value here intended?

Reply:
yes, was having problem on preview and staging where end request is http not https...

Reply:
If preview and staging are not https then the secure flag should be set in those environments. Listening to the requests preference on secure or not will allow users to access and interact with the service on non-secure http connections where you expect https connections. 

This violates the principle of secure by default and I'd recommend not doing it.

Reply:
have just tested this, unfortunately again we cant do this... The external url is secure, but the applications internal url is not. therefor setting this explicitly isn't going to work ( if we want to test our applications in the pipeline ), Except for maybe production....

I will edit this so req.secure is the default value so it can be overridden